packages/renderer/src/lib/deployments/DeploymentEmptyScreen.svelte

```diff
@@ -1,7 +1,18 @@
 <script lang="ts">
+import { onMount } from 'svelte';
+
 import DeploymentIcon from '../images/DeploymentIcon.svelte';
 import KubernetesEmptyScreen from '../kube/KubernetesEmptyScreen.svelte';
+import { listenResourcePermitted } from '../kube/resource-permission';
+
+let deploymentsPermitted: boolean = $state(true);
+
+onMount(() => {
+  return listenResourcePermitted('deployments', (permitted: boolean) => {
```

Comment:
Finally, returning the result of the `listen...` does not work as we expected when working on this code, because the value is a Promise and not synchronous:

```
If a function is returned synchronously from onMount, it will be called when the component is unmounted.
```

You will need to save the disposable and call it from `onDestoy`